### PR TITLE
MVP - Employ block verification flow in NaiveTransitionFrontier

### PR DESCRIPTION
**Pull request by hanabi1224**
**Summary of changes**
Changes introduced in this pull request:
- Verify blocks in the `NaiveTransitionFrontier` with the logic implemented in https://github.com/ChainSafe/mina-rs/pull/238
- Fix `Hashable` implementation for `StakeDelegation`

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

### CODE REVIEW

**Review by sudo-shashank — APPROVED** [review]

**Review by creativcoder — APPROVED** [review]

File: base/tests/verification.rs [hunk 1]
```diff
@@ -6,16 +6,30 @@ mod tests {
...
         let mut ctx = mina_signer::create_legacy::<SignedCommandPayload>(NetworkId::MAINNET);
 
         assert!(TEST_BLOCKS.iter().all(|(_, v)| {
             let block = ExternalTransition::from(v.external_transitionv1().unwrap());
             block.verify(&mut ctx)
         }))
     }
+
+    #[test]
+    fn verify_all_json_fixtures() {
```
Comment (willemolding) [inline base/tests/verification.rs:24]: Thanks for adding this one

File: network/src/processor/naive_transition_frontier.rs [hunk 2]
```diff
@@ -38,14 +39,19 @@ impl TransitionFrontier for NaiveTransitionFrontier {
     }
 
     async fn add_block(&mut self, block: Self::Block) -> anyhow::Result<()> {
-        if self.best_chain.length() < 1 {
-            self.best_chain.push(block.protocol_state)?;
+        let mut ctx = mina_signer::create_legacy::<SignedCommandPayload>(NetworkId::MAINNET);
```
Comment (willemolding) [inline network/src/processor/naive_transition_frontier.rs:42]: This crate might be an appropriate place to make this a `static mut` somewhere. Not sure about this though

File: base/src/user_commands/signed_command/mod.rs [hunk 3]
```diff
@@ -110,10 +111,26 @@ impl Hashable for SignedCommandPayload {
...
+                    roi.append_bytes(&self.common.memo.0);
+
+                    for tag_bit in DELEGATION_TX_TAG {
+                        roi.append_bool(tag_bit);
+                    }
+
+                    roi.append_bool(delegator.is_odd);
                     roi.append_bool(new_delegate.is_odd);
+                    roi.append_u64(1);
+                    roi.append_u64(0);
```
Comment (willemolding) [inline base/src/user_commands/signed_command/mod.rs:132]: What are these fields?

**Review by willemolding — APPROVED** [review]

File: base/src/user_commands/signed_command/mod.rs (hunk 3 above)
Comment (hanabi1224) [reply base/src/user_commands/signed_command/mod.rs:132]: 1 is `TokenId` and 0 is `Amount`

File: network/src/processor/naive_transition_frontier.rs (hunk 2 above)
Comment (hanabi1224) [reply network/src/processor/naive_transition_frontier.rs:42]: To reuse ctx in a thread-safe way, I would use an object pool here

**Pull request merged**